napari/layers/base/base.py

```diff
@@ -998,6 +1004,123 @@ def _data_to_world(self) -> Affine:
         """
         return self._transforms[1:3].simplified
 
+    def vector_world_to_data(self, vector) -> tuple:
+        """Convert a vector defining an orientation from world coordinates to data coordinates.
+        For example, this would be used to convert the view ray.
+
+        Parameters
+        ----------
+        vector : tuple, list, 1D array
+            A vector in world coordinates.
+
+        Returns
+        -------
+        tuple
+            Vector in data coordinates.
+        """
+        vector = np.asarray(vector)
+
+        # create transform for view direction (world -> layer data)
+        shear_matrix = expand_upper_triangular(self.shear)
+        rot_matrix = self.rotate
+        scale_matrix = np.diag(self.scale)
+        world_to_layer = np.linalg.inv(
+            shear_matrix @ rot_matrix
+        ).T @ np.linalg.inv(scale_matrix)
+
+        return tuple(vector @ world_to_layer)
+
+    @property
+    def _display_bounding_box(self):
+        """An axis aligned (self._ndisplay, 2) bounding box around the data"""
+        return self._extent_data[:, self._dims_displayed].T
+
+    def _data_face_intersection(
+        self, vertices: np.ndarray, world_to_canvas, click_pos_canv
+    ):
+
+        # convert the vertex coordinates to world coordinates
+        vertices_world = self._data_to_world(vertices)
+
+        # convert vertex world to canvas coordinates
+        vertices_canv = world_to_canvas.map(
+            np.asarray(vertices_world)[:, [2, 1, 0]]
+        )
+        vertices_canv = vertices_canv[:, :2] / vertices_canv[:, 3:]
+        triangle_vertices_canv = np.stack(
+            (vertices_canv[[0, 1, 2]], vertices_canv[[0, 2, 3]])
+        )
+        in_triangles = inside_triangles(
```

Comment:
@alisterburt , I think if we do the following, we don't need the canvas to world transform:

        # 1) dont' convert to canvas
        # 2) sub. inside_triangles for 3D version
        # 4) do detection there